### ISSUE TITLE
Replace TensorBoard audio with Kokoro-faithful inference and Stage 1 baseline comparison

### ISSUE THREAD

**Issue by semidark** [semidark/kikiri-tts#3]
### Problem

The current TensorBoard audio generation in `train_second.py` uses a completely different pipeline than the actual Kokoro inference, making the audio samples misleading:

- **Before `joint_epoch`:** Uses ground-truth alignment, ground-truth style from GT mel spectrograms, and tests decoder reconstruction in isolation. Sounds like static noise if the decoder is weak, even when real inference would produce speech.
- **After `joint_epoch`:** Uses the diffusion sampler to generate style predictions — but diffusion is disabled in our config (`diff_epoch=999`), so the sampler produces garbage.

Neither mode reflects what `extract_voicepack.py` + `test_inference.py` (i.e., real Kokoro inference) actually produces. This forces you to stop training, extract a voicepack, and run inference manually just to evaluate model quality — a process that takes minutes and breaks flow.

### Proposed Solution

Replace both TensorBoard audio generation blocks with Kokoro-faithful inference, and add a **Stage 1 baseline** for direct comparison.

#### Stage 1 Baseline (logged once at epoch 0, before any training updates)

At the start of Stage 2 training, before the first `optimizer.step()`, the model weights are exactly the Stage 1 output. We use this moment to:

1. **Pre-extract a voicepack** from 200 randomly sampled audio files using the Stage 1 `style_encoder` and `predictor_encoder` — same logic as `extract_voicepack.py`.
2. **Run Kokoro-faithful inference** on 7 hardcoded German test sentences (the same phonetic test set from `test_inference.py`) using the Stage 1 model weights.
3. **Log as `baseline/test_01` through `baseline/test_07`** — these never change and serve as the "Stage 1 could do this" reference.
4. **Log `baseline/acoustic_norm` and `baseline/prosodic_norm`** as reference scalar values.

#### Per-Epoch Inference (logged at the end of every epoch)

1. **Re-extract the voicepack** from the same 200 audio samples using the current (potentially updated) `style_encoder` and `predictor_encoder`.
2. **Run Kokoro-faithful inference** on the same 7 test sentences using current model weights:
   - Predict duration from the predictor
   - Build alignment from predicted duration
   - Predict F0 and energy
   - Decode audio using the voicepack's acoustic style
3. **Log as `inference/test_01` through `inference/test_07`**.
4. **Log `voicepack/acoustic_norm` and `voicepack/prosodic_norm`** as scalar graphs to monitor `style_encoder` health across epochs.

#### "Kokoro-faithful inference" means

The TensorBoard audio generation mirrors `KModel.forward_with_tokens` exactly:

```python
input_ids = [0] + token_ids + [0]                          # BOS/EOS
bert_dur = model.bert(input_ids, ...)
d_en = model.bert_encoder(bert_dur).transpose(-1, -2)
s = voicepack_prosodic                                      # from voicepack, not GT mel
d = model.predictor.text_encoder(d_en, s, ...)
x, _ = model.predictor.lstm(d)
duration = torch.sigmoid(model.predictor.duration_proj(x)).sum(axis=-1)
pred_dur = torch.round(duration).clamp(min=1)
pred_aln_trg = <build alignment from pred_dur>
en = d.transpose(-1, -2) @ pred_aln_trg
F0_pred, N_pred = model.predictor.F0Ntrain(en, s)
t_en = model.text_encoder(input_ids, ...)
asr = t_en @ pred_aln_trg
audio = model.decoder(asr, F0_pred, N_pred, voicepack_acoustic)
```

No ground-truth alignment, no ground-truth pitch, no live style computation from GT mels. What you hear is what you get.

### TensorBoard Layout

| Tag | Description | Logged |
|---|---|---|
| `baseline/test_01` – `baseline/test_07` | Stage 1 Kokoro inference (fixed reference) | Epoch 0 only |
| `baseline/acoustic_norm` | Stage 1 style encoder output norm | Epoch 0 only |
| `baseline/prosodic_norm` | Stage 1 predictor encoder output norm | Epoch 0 only |
| `inference/test_01` – `inference/test_07` | Current Stage 2 Kokoro inference | Every epoch |
| `voicepack/acoustic_norm` | Current style encoder norm (scalar graph) | Every epoch |
| `voicepack/prosodic_norm` | Current predictor encoder norm (scalar graph) | Every epoch |

### Benefits

- **What you hear is what you get:** TensorBoard audio matches real `extract_voicepack.py` + `test_inference.py` output
- **Direct A/B comparison:** Stage 1 baseline vs Stage 2 current, same sentences, same pipeline — hear the improvement (or degradation) in real time
- **Early stopping signal:** If Stage 2 audio degrades relative to Stage 1, stop training immediately instead of wasting hours
- **Style encoder monitoring:** Voicepack norm scalars provide an early warning if the encoder collapses (norm dropping from ~2.2 toward ~0.37)
- **Consistent test set:** Same 7 German sentences every epoch, directly comparable across runs and experiments

### Test Sentences

From `scripts/test_inference.py`:

1. "Schön, dass du da bist. Die Bücher liegen auf dem großen Tisch." — Umlauts, sch
2. "Ich mache mich auf den Weg nach Aachen, um auch nachts wach zu sein." — Ich-Laut vs Ach-Laut
3. "Er aß die Maße in der Straße, aber das Maß war voll." — Eszett, vowel length
4. "Zwei weiße Zwerge zwängen sich zwischen zwei Zweige." — Zischlaute, consonant clusters
5. "Ein Pfau pflegt seine Federn an der Pfütze." — Pf-Laute
6. "Warum hast du das getan? Das ist ja unglaublich!" — Prosody: questions and exclamations
7. "Das kostet genau einhundertdreiundzwanzig Millionen Euro." — Numbers

### Files Changed

- `StyleTTS2/train_second.py` — Replace TensorBoard audio blocks with Kokoro-faithful inference. Add voicepack pre-extraction and test sentence preparation before training loop.

### Dependencies (all already installed)

- `misaki` with espeak backend — for German G2P
- `soundfile` — for WAV loading during voicepack extraction  
- `torchaudio` — for mel spectrogram computation
- `espeak-ng` — system dependency for German phonemization

**Comment by semidark** [comment]
First training run with the improved Tensorboard audiosamples is currently running.
